[caffe2] Export operators to c10 without including ATen/Tensor.h

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #67096
* #67094

With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

peterbell10 committed:
[caffe2] Export operators to c10 without including ATen/Tensor.h

With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289810).

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289810).

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289810).

dagitses commented on caffe2/core/export_caffe2_op_to_c10.h:
```diff
@@ -36,117 +36,31 @@ inline std::vector<caffe2::Tensor> _call_caffe2_op(
-#endif
-}
+struct TORCH_API RegisterSchema {
+  RegisterSchema(const char *schema_str);
```
are you able to mark this explicit?

peterbell10 replied:
done

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Rebase after master CI break on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289810).

peterbell10 committed:
Only build IValueInterface when necessary on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289810).

peterbell10 committed:
Rebase and fix merge conflicts on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Fix bad merge on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Rebase on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Rebase on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32289810).

kit1980 commented:
@peterbell10 @dagitses is this something we want to merge now?

peterbell10 commented:
There is still a caffe2 job in PyTorch CI so I think it still has value.

kit1980 commented:
@pytorchbot rebase

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]

peterbell10 committed:
Update on "[caffe2] Export operators to c10 without including ATen/Tensor.h"


With this change, the only caffe2 files that depend on `ATen/Tensor.h`
are ones that directly use the ATen API. Specifically,
```
[
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/contrib/aten/aten_op_gpu.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/tensor.cc.o",
  "caffe2/CMakeFiles/torch_cuda_cpp.dir/operators/layer_norm_op.cu.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/IValueInterface.cc.o",
  "caffe2/CMakeFiles/cuda_tensor_interop_test.dir/__/aten/src/ATen/test/cuda_tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/torch_cpu.dir/contrib/aten/aten_op.cc.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state_gpu.dir/python/pybind_state.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/layer_norm_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_c10_op_to_caffe2.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/export_caffe2_op_to_c10.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/operators/enforce_finite_op.cc.o",
  "caffe2/CMakeFiles/torch_cpu.dir/core/operator.cc.o",
  "caffe2/CMakeFiles/tensor_interop_test.dir/__/aten/src/ATen/test/tensor_interop_test.cpp.o",
  "caffe2/CMakeFiles/caffe2_pybind11_state.dir/python/pybind_state.cc.o"
]
```

Differential Revision: [D32289810](https://our.internmc.facebook.com/intern/diff/D32289810)

[ghstack-poisoned]
